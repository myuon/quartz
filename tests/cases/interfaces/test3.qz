interface Output {
    fun gen1(self): i32;
    fun gen2(self): i32;
    fun gen3(self): i32;
    fun gen4(self): i32;
}

struct Impl1 {
}

module Impl1: Output {
    fun gen4(self): i32 {
        return 31;
    }

    fun gen2(self): i32 {
        return 23;
    }

    fun gen1(self): i32 {
        return 17;
    }

    fun gen3(self): i32 {
        return 29;
    }
}

fun main() {
    let v = make[vec[Output]]();
    v.push(Impl1{});

    println("{} {} {} {}", v.(0).gen1().to_string(), v.(0).gen2().to_string(), v.(0).gen3().to_string(), v.(0).gen4().to_string());
}
