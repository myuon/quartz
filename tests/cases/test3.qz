fun calc(b: i32): i32 {
    let a = 1;
    let z = 10;
    return z + a + b;
}

fun main(): i32 {
    return calc(2);
}

