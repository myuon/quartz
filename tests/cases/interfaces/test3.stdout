17 23 29 31
